app/cdap/components/Connections/Browser/SidePanel/CategorizedConnections.tsx

```diff
@@ -156,6 +183,105 @@ export function CategorizedConnections({
     }
   };
 
+  function getConnectionConfig(conn) {
+    const connectionConfig = {
+      name: conn.name,
+      description: conn.description,
+      category: conn.plugin.category,
+      plugin: conn.plugin,
+    };
+
+    return connectionConfig;
+  }
+
+  function exportConnection(conn) {
+    const connectionConfig = getConnectionConfig(conn);
+    const artifact = objectQuery(conn, 'plugin', 'artifact') || {};
+    DownloadFile(
+      connectionConfig,
+      null,
+      `${conn.name}-connector-${artifact.name}-${artifact.version}`
+    );
+  }
+
+  function cloneConnection(conn) {
+    const config = getConnectionConfig(conn);
+    delete config.name;
+
+    openConnectionConfig(config);
+  }
+
+  function openConnectionConfig(config) {
+    setInitConnConfig(config);
+    toggleConnectionCreate();
+  }
+
+  function editConnection(conn) {
+    const config = getConnectionConfig(conn);
+    setIsEdit(true);
+    openConnectionConfig(config);
+  }
+
+  function handleConfirmationClose() {
+    setConnectionToDelete(null);
+    setDeleteError(null);
+  }
+
+  function toggleConnectionCreate() {
+    const newState = !isCreateConnectionOpen;
+
+    setIsCreateConnectionOpen(newState);
+
+    if (!newState) {
+      setInitConnConfig(null);
+      setIsEdit(false);
+    }
+  }
+
+  function handleDelete() {
+    if (!connectionToDelete) {
+      return;
+    }
+
+    const params = {
+      context: getCurrentNamespace(),
+      connectionId: connectionToDelete.name,
+    };
+
+    return ConnectionsApi.deleteConnection(params).subscribe(
+      () => {
+        handleConfirmationClose();
+        fetchConnections();
+      },
+      (err) => {
+        setDeleteError(err);
+      }
+    );
+  }
+
+  const popperModifiers = {
+    preventOverflow: {
+      enabled: true,
+      boundariesElement: objectQuery(boundaryElement, 'current'),
+    },
+    hide: {
+      enabled: false,
+    },
+  };
+
+  let confirmDeleteElem;
+  if (connectionToDelete) {
+    confirmDeleteElem = (
+      <div>
+        Are you sure you want to delete{' '}
```

Comment:
Is translation needed?

Reply:
we haven't migrate connection management to use i18n yet